lib/components/tweet/widgets/media/tweet_images.dart

```diff
@@ -66,6 +54,62 @@ class _TweetImagesState extends State<TweetImages> {
     );
   }
 
+  Future<void> _onImageLongPress(int index, BuildContext context) async {
+    _galleryIndex = index;
+
+    await showModalBottomSheet<void>(
+      context: context,
+      shape: const RoundedRectangleBorder(
+        borderRadius: BorderRadius.only(
+          topLeft: kDefaultRadius,
+          topRight: kDefaultRadius,
+        ),
+      ),
+      builder: (BuildContext context) => Column(
+        mainAxisSize: MainAxisSize.min,
+        children: <Widget>[
+          const ModalSheetHandle(),
+          ListTile(
+            leading: const Icon(FeatherIcons.share),
+            title: const Text('open externally'),
+            onTap: _onOpenImageExternaly,
+          ),
+          ListTile(
+            leading: const Icon(FeatherIcons.download),
+            title: const Text('download'),
+            onTap: _onDownloadImage,
+          ),
+          ListTile(
+            leading: const Icon(FeatherIcons.share2),
+            title: const Text('share'),
+            onTap: _onShareImage,
+          ),
+        ],
+      ),
```

Comment:
Since we build the same column 3 times we should add an extra widget for it where we just pass along the callbacks.
Can be in `lib\components\tweet\widgets\media`